### PR TITLE
Fix issue with forward-delete in editable text widget

### PR DESCRIPTION
**Pull request by krisgiesing**
Fixes index issues when a forward delete is attempted at the end of the text range. Logic now mirrors similar checks done at the beginning of the text range.

### CODE REVIEW

File: packages/flutter/lib/src/widgets/editable_text.dart [hunk 1]
```diff
@@ -52,7 +52,7 @@ class EditableString implements KeyboardClient {
   }
 
   String textAfter(TextRange range) {
-    return text.substring(range.end);
+    return text.substring(range.end, text.length);
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/editable_text.dart]: won't that be guaranteed to go past the end?

Comment (krisgiesing) [inline packages/flutter/lib/src/widgets/editable_text.dart]: Hm, let me re-examine that change.

**Comment by Hixie** [comment]
LGTM without the textAfter change.

File: packages/flutter/lib/src/widgets/editable_text.dart (hunk 1 above)
Comment (krisgiesing) [inline packages/flutter/lib/src/widgets/editable_text.dart]: End is exclusive, but you're right this is the default behavior. So the change is harmless but irrelevant.

**Pull request merged**